test/e2e/cluster_extension_registryV1_limitations_test.go

```diff
@@ -2,60 +2,36 @@ package e2e
 
 import (
 	"context"
-	"os"
 	"testing"
 
 	"github.com/stretchr/testify/assert"
 	"github.com/stretchr/testify/require"
-	"k8s.io/apimachinery/pkg/api/errors"
 	apimeta "k8s.io/apimachinery/pkg/api/meta"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/types"
 
-	catalogd "github.com/operator-framework/catalogd/api/core/v1alpha1"
-
 	ocv1alpha1 "github.com/operator-framework/operator-controller/api/v1alpha1"
 )
 
 func TestClusterExtensionPackagesWithWebhooksAreNotAllowed(t *testing.T) {
 	ctx := context.Background()
-	catalog, err := createTestCatalog(ctx, testCatalogName, os.Getenv(testCatalogRefEnvVar))
-	defer func(cat *catalogd.Catalog) {
-		require.NoError(t, c.Delete(context.Background(), cat))
-		require.Eventually(t, func() bool {
-			err := c.Get(context.Background(), types.NamespacedName{Name: cat.Name}, &catalogd.Catalog{})
-			return errors.IsNotFound(err)
-		}, pollDuration, pollInterval)
-	}(catalog)
-	require.NoError(t, err)
-
-	deleteClusterExtension := func(clusterExtension *ocv1alpha1.ClusterExtension) {
-		require.NoError(t, c.Delete(ctx, clusterExtension))
-		require.Eventually(t, func() bool {
-			err := c.Get(ctx, types.NamespacedName{Name: clusterExtension.Name}, &ocv1alpha1.ClusterExtension{})
-			return errors.IsNotFound(err)
-		}, pollDuration, pollInterval)
-	}
+	clusterExtension, catalog := testInit(t)
 
-	clusterExtension := &ocv1alpha1.ClusterExtension{
-		ObjectMeta: metav1.ObjectMeta{
-			GenerateName: "package-with-webhooks-",
-		},
-		Spec: ocv1alpha1.ClusterExtensionSpec{
-			PackageName:      "package-with-webhooks",
-			Version:          "1.0.0",
-			InstallNamespace: "default",
-		},
+	clusterExtension.Spec = ocv1alpha1.ClusterExtensionSpec{
+		PackageName:      "package-with-webhooks",
+		Version:          "1.0.0",
+		InstallNamespace: "default",
 	}
-	err = c.Create(ctx, clusterExtension)
-	defer deleteClusterExtension(clusterExtension)
-	require.NoError(t, err)
+	require.NoError(t, c.Create(ctx, clusterExtension))
+	defer testCleanup(t, catalog, clusterExtension)
```

Comment:
If there is an error on line 25 `testCleanup` will not be called. It is fine in CI, but this can be annoying locally if you try to re-use kind cluster for repeated test execution.

Reply:
We'll fix this to be the right pattern.
OWNER: @dtfranz

Reply:
Fixing in #883

Reply:
Done!